plugins/scaffolder-backend/src/scaffolder/actions/TemplateActionRegistry.ts

```diff
@@ -16,13 +16,39 @@
 
 import { ConflictError, NotFoundError } from '@backstage/errors';
 import { TemplateAction } from '@backstage/plugin-scaffolder-node';
+import { ActionsService } from '@backstage/backend-plugin-api/alpha';
+import {
+  AuthService,
+  BackstageCredentials,
+  LoggerService,
+} from '@backstage/backend-plugin-api';
+import { isPlainObject } from 'lodash';
+import { Schema } from 'jsonschema';
+import { JsonObject } from '@backstage/types';
+
+/**
+ * @internal
+ */
+export interface TemplateActionRegistry {
+  register(action: TemplateAction<any, any, any>): void;
+  get(actionId: string): Promise<TemplateAction<any, any, any>>;
```

Comment:
Thinking that `get` should also have `{ credentials }`. It's going to lead to some funky behavior possibly when permissions comes along.

That way I think we can cleanup the `await auth.getOwnServiceCredentials()` everywhere as they should all be request driven AFAIK.

Reply:
Hm the `NunjucksWorkflowRunner` already has access to this though.

`this.options.actionRegistry.get(step.action, { credentials: task.getInitiatorCredentials())` should do it.

Reply:
Damn, missed that! I will use it